src/ui/map.js

```diff
@@ -1987,6 +1996,33 @@ class Map extends Camera {
             this.style._releaseSymbolFadeTiles();
         }
 
+        if (this.listens('gpu-timing-frame')) {
+            const renderCPUTime = browser.now() - frameStartTime;
+            extTimerQuery.endQueryEXT(extTimerQuery.TIME_ELAPSED_EXT, gpuTimer);
+            setTimeout(() => {
+                const renderGPUTime = extTimerQuery.getQueryObjectEXT(gpuTimer, extTimerQuery.QUERY_RESULT_EXT) / (1000 * 1000);
+                extTimerQuery.deleteQueryEXT(gpuTimer);
+                this.fire(new Event('gpu-timing-frame', {
+                    cpuTime: renderCPUTime,
+                    gpuTime: renderGPUTime
+                }));
+            }, 50); // Wait 50ms to give time for all GPU calls to finish before querying
```

Comment:
It's not clear how the threshold here works — is 50ms an arbitrary value that means "long enough to be sure"? Is this because `endQueryExt` happens async too, without any way to track when it executes?